kube_apiserver_metrics/datadog_checks/kube_apiserver_metrics/kube_apiserver_metrics.py

```diff
@@ -0,0 +1,134 @@
+# (C) Datadog, Inc. 2019
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import os
+from copy import deepcopy
+
+from six import iteritems
+from datadog_checks.base import ConfigurationError
+from datadog_checks.checks.openmetrics import OpenMetricsBaseCheck
+
+
+class KubeApiserverMetricsCheck(OpenMetricsBaseCheck):
+    """
+    Collect kubernetes apiserver metrics in the Prometheus format
+    See https://github.com/kubernetes/apiserver
+    """
+    DEFAULT_METRIC_LIMIT = 0
+    # Set up metric_transformers
+    METRIC_TRANSFORMERS = {}
+    DEFAULT_BEARER_TOKEN_PATH = '/var/run/secrets/kubernetes.io/serviceaccount/token'
+    DEFAULT_SCHEME = 'https'
+
+    def __init__(self, name, init_config, agentConfig, instances=None):
+        # Set up metric_transformers
+        self.METRIC_TRANSFORMERS = {
+                'apiserver_audit_event_total': self.apiserver_audit_event_total,
+                'rest_client_requests_total': self.rest_client_requests_total,
+                'apiserver_request_count': self.apiserver_request_count,
+                'apiserver_dropped_requests_total': self.apiserver_dropped_requests_total,
+                'http_requests_total': self.http_requests_total,
+                'authenticated_user_requests': self.authenticated_user_requests,
+        }
+        self.bearer_token_found = False
+        # Create instances we can use in OpenMetricsBaseCheck
+        generic_instances = None
+        if instances is not None:
+            generic_instances = self.create_generic_instances(instances)
```

Comment:
Let's not change the users' instances but rather have a `self.kube_apiserver_config = None` and set it on the first check run

Reply:
ok that's even better actually as I can add logging from the agent class.